src/components/__tests__/RouteItemEdit.cy.js

```diff
@@ -0,0 +1,265 @@
+    cy.testLanguageStringsInContext(
+      [
+        'labelDistance',
+        'labelDirectionFromWork',
+        'labelDirectionToWork',
+        'labelTransportType',
+      ],
+      'routes',
+      i18n,
+    );
+  });
+
+  context('route to work', () => {
+    beforeEach(() => {
+      cy.fixture('routeListItem').then((routes) => {
+        cy.mount(RouteItemEdit, {
+          props: {
+            route: routes.toWork,
+          },
+        });
+        cy.viewport('macbook-16');
+      });
+    });
+
+    coreTests();
+
+    it('renders label direction', () => {
+      // direction label
+      cy.dataCy('column-direction').should('be.visible');
+      cy.dataCy('label-direction').should(
+        'contain',
+        i18n.global.t('routes.labelDirectionToWork'),
+      );
+      // direction icon
+      cy.dataCy('label-direction-icon').should('be.visible');
+      cy.dataCy('label-direction-icon').should('contain', 'arrow_forward');
+    });
+
+    it('renders the input distance value', () => {
+      // make sure transport type is bike
+      cy.dataCy('button-toggle-transport').find('button').eq(0).click();
+      cy.fixture('routeListItem').then((routes) => {
+        // input distance value
+        cy.dataCy('input-distance')
+          .should('be.visible')
+          .invoke('val')
+          .should('eq', routes.toWork.distance);
+      });
+    });
+
+    it('renders correct transport type', () => {
+      cy.fixture('routeListItem').then((routes) => {
+        cy.dataCy('description-transport')
+          .should('be.visible')
+          .and(
+            'contain',
+            i18n.global.t(`routes.transport.${routes.toWork.transport}`),
+          );
+      });
+    });
+  });
+
+  context('route from work', () => {
+    beforeEach(() => {
+      cy.fixture('routeListItem').then((routes) => {
+        cy.mount(RouteItemEdit, {
+          props: {
+            route: routes.fromWork,
+          },
+        });
+        cy.viewport('macbook-16');
+      });
+    });
+
+    coreTests();
+
+    it('renders label direction', () => {
+      // direction label
+      cy.dataCy('column-direction').should('be.visible');
+      cy.dataCy('label-direction').should(
+        'contain',
+        i18n.global.t('routes.labelDirectionFromWork'),
+      );
+      // direation icon
+      cy.dataCy('label-direction-icon').should('be.visible');
+      cy.dataCy('label-direction-icon').should('contain', 'arrow_back');
+    });
+
+    it('renders "0" when distance is empty', () => {
+      // empty distance value
+      cy.dataCy('input-distance')
+        .should('be.visible')
+        .invoke('val')
+        .should('eq', '0');
+    });
+  });
+
+  context('mobile', () => {
+    beforeEach(() => {
+      cy.fixture('routeListItem').then((routes) => {
+        cy.mount(RouteItemEdit, {
+          props: {
+            route: routes.toWork,
+          },
+        });
+        cy.viewport('iphone-6');
```

Comment:
Do you test how is layout rendered (mobile viewport vs desktop viewport)?

Reply:
Did not test this before. Added.